packages/animation-curve-block/src/components/AnimationCanvas.tsx

```diff
@@ -105,18 +105,27 @@ export const AnimationCanvas = ({
             };
 
             const animationValues = {
-                x: positionWithinViewBox.x / viewBox.width,
+                x: calculateXValue(positionWithinViewBox.x, viewBox.width),
                 y: 1 - positionWithinViewBox.y / viewBox.height,
             };
 
-            if (animationValues.x >= 0 && animationValues.x <= 1) {
-                setAnimationFunction && setAnimationFunction(updatedAnimationFunction(animationValues, draggingPoint));
-            }
+            setAnimationFunction && setAnimationFunction(updatedAnimationFunction(animationValues, draggingPoint));
         },
         // eslint-disable-next-line react-hooks/exhaustive-deps
         [draggingPoint, viewBox.width, viewBox.height]
     );
 
+    const calculateXValue = (viewBoxX: number, viewBoxWidth: number) => {
+        const x = viewBoxX / viewBoxWidth;
+        if (x < 0) {
```

Comment:
Optional, to consume less lines:
`return Math.max(Math.min(x, 1), 0)`

Reply:
Thanks. Adjusted and got rid of that function